src/main/java/seedu/address/ui/Clock.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.ui;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import javafx.animation.KeyFrame;
+import javafx.animation.Timeline;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.control.SplitPane;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.util.Duration;
+import seedu.address.commons.util.FxViewUtil;
+
+//@@author A0135998H
+public class Clock extends UiPart<Region> {
+    private static final String FXML = "Clock.fxml";
+    private static final int CLOCK_RENDERING_DELAY = 500; // msec
+
+    private DateTimeFormatter formatter;
+
+    @FXML
+    private Label clockLabel;
+
+    public Clock(AnchorPane clockPlaceholder) {
+        super(FXML);
+        formatter = DateTimeFormatter.ofPattern("d MMM EEEE, h:mm a");
+        addToPlaceholder(clockPlaceholder);
+        clockLabel.setText(formattedDate());
```

Comment:
`formattedDate()` is funny. `formatDate()` might be better but since it is the current time, shouldn't `getCurrentFormattedDate()` be better? Also the formatting probably should not be done in this class (SRP).